src/Resolvers/CastResolver.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Spatie\DataTransferObject\Resolvers;
+
+use Spatie\DataTransferObject\Attributes\CastWith;
+use Spatie\DataTransferObject\Descriptors\ClassDescriptor;
+use Spatie\DataTransferObject\Descriptors\PropertyDescriptor;
+
+class CastResolver
+{
+    public function __construct(private ClassDescriptor $descriptor)
+    {
+        //
+    }
+
+    public function castArguments(array $arguments): array
+    {
+        foreach ($arguments as $argument => $value) {
+            $arguments[$argument] = $this->castArgument(
+                $this->descriptor->getPropertyByName($argument),
+                $value
+            );
+        }
+
+        return $arguments;
+    }
+
+    protected function castArgument(PropertyDescriptor $property, mixed $value): mixed
+    {
+        $attributes = $property->getReflection()->getAttributes(CastWith::class);
```

Comment:
We can also use spatie/better-types here. 

I'm also fine with being able to resolve the cast information from without the property descriptor to keep this code a little more readble.

eg `$property->getCastWith` or something alike. 

I think our custom attributes are part of the "schema" or "description" of the property, it's only what we do with it that should be handled by this class. Let me know your thoughts though!

Reply:
Agreed. I think it might be better to add a `getAttribute` method (utilizing `spatie/better-types`) so that a handy helper for other Attributes is present as well.